File: bai5andanh/app/src/main/java/com/example/bai5andanh/MainActivity.java
package com.example.bai5andanh;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    EditText A, B;
    Button cong;
    TextView KQ;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        // Ánh xạ ID từ layout
        A = findViewById(R.id.input1);
        B = findViewById(R.id.input2);
        cong = findViewById(R.id.btncong);
        KQ = findViewById(R.id.kq);

        // Gán sự kiện bấm nút
        cong.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.btncong) {
            try {
                int soA = Integer.parseInt(A.getText().toString());
                int soB = Integer.parseInt(B.getText().toString());
                int tong = soA + soB;
                KQ.setText("Kết quả: " + tong);
            } catch (NumberFormatException e) {
                Toast.makeText(this, "Vui lòng nhập số hợp lệ!", Toast.LENGTH_SHORT).show();
            }
        }
    }
}

File: ex4_congtrunhanchia/app/src/main/java/com/example/ex4_congtrunhanchia/MainActivity.kt
package com.example.ex4_congtrunhanchia

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        val a= findViewById<EditText>(R.id.input1)
        val b= findViewById<EditText>(R.id.input2)
        val BtnCong= findViewById<Button>(R.id.btncong)
        val BtnTru= findViewById<Button>(R.id.btntru)
        val BtnNhan= findViewById<Button>(R.id.btnn)
        val BtnChia= findViewById<Button>(R.id.btnchia)
        val KQ= findViewById<TextView>(R.id.kq)
        BtnCong.setOnClickListener{
            TinhTong(a, b, KQ)
        }


    }
    private fun TinhTong(a:EditText,b:EditText,KQ:TextView){
        val soa=a.text.toString().toInt()
        val sob=b.text.toString().toInt()
        val sum=soa+sob
        KQ.text="kết quả cộng là: $sum"
    }
}